File: dcp-solutions/2024-11-18/solution.rb
```ruby
def max_product(nums)
  nums.sort!

  # Product of three largest numbers
  product1 = nums.last * nums[-2] * nums[-3]

  # Product of two smallest numbers and one largest number
  product2 = nums.first * nums[1] * nums.last

  # Return the maximum of the two products
  [product1, product2].max
end
```

File: dcp-solutions/2024-11-18/solution.py
```python
def largest_product(nums):
    nums.sort()
    n = len(nums)
    
    # Handle the case where there are four negative numbers and one positive number.
    if nums[n-1] < 0:
        return max(nums[-1]*nums[-2]*nums[-3], nums[0]*nums[1]*nums[2])
    
    # Calculate the product of three largest numbers
    return nums[n-1]*nums[n-2]*nums[n-3]
```